### ISSUE TITLE
Android crashes with "Invalid notification (no valid small icon)"

### ISSUE THREAD

**Issue by shinyford** [MaikuB/flutter_local_notifications#73]
Hi 

I'm having problems configuring this for Android. Whatever I do, I can't seem to identify the icon correctly, meaning every notification throws the following crash:

```java.lang.RuntimeException: Unable to start receiver com.dexterous.flutterlocalnotifications.ScheduledNotificationReceiver: java.lang.IllegalArgumentException: Invalid notification (no valid small icon): Notification(channel=FEA_NOTIFICATION pri=1 contentView=null vibrate=null sound=null defaults=0x0 flags=0x10 color=0x00000000 vis=PRIVATE)```

Android setting are configured thus:

 ```final initializationSettingsAndroid = AndroidInitializationSettings('<ICON NAME>');```

And I've tried every combination I can think of for ICON_NAME: `app_icon`, `ic_launcher`, 'ic_launcher.png`, '@mipmap/ic_launcher` and so on.

I wonder if I'm just doing something stupid? Is there something obvious I'm missing?

Many thanks

Nic

It all works beautifully on iOS, BTW - many thanks for this.

**Comment by MaikuB** [comment]
Yes as per the readme it should be a drawable resource and I've also mentioned that there's a example project for the plugin. If the launcher one isn't working anymore then perhaps Flutter has changed their APIs so that it won't be found anymore. Frankly I'm surprised others found that it worked. Also, the mipmap is for the launcher icon and not intended to be used for the notification/status icon. The name you pass for the icon to the plugin isn't arbitrary either. Links have already been provided on how Android's resource system works. I suggest you check the readme, links and example project again

**Comment by shinyford** [comment]
I'll check again - but it wasn't clear to me. So, in the example project you pass in the string `app_icon` - but that doesn't seem to be part of the standard Android file system. I wasn't sure if you meant that string to be used verbatim (it causes the crash, BTW) or if it was a synonym for something. (For example, elsewhere in the app you use `your channel id` as the channel id, so it looks like those strings might be placeholders rather than actual strings that could be used.)

FWIW I have created a drawable resource, `notification_icon.png` and tried to use a string that references that: it still crashes, whether I pass the string `notification_icon` or `notification_icon.png` into `AndroidInitializationSettings`.

**Comment by b-connect** [comment]
```
new AndroidInitializationSettings('@mipmap/ic_launcher');
```
uses default icon

**Comment by MaikuB** [comment]
Are you saying you have an issue with that or was that to help show @shinyford how to use to launcher icon?
